File: frontend/src/api/dlpPolicySearch.ts
import apiClient from './client'

export interface DlpPolicyExclusion {
  object_id: string | number | null
  object_name: string | null
  object_description: string | null
  object_status: string | null
  rule_type: string | number | null
  used_as: 'SENDER' | 'RECIPIENT' | 'UNUSED' | string
  policy_id: string | number | null
  policy_name: string | null
  policy_active_status: string | number | null
  policy_record_status: string | null
  user_patterns: string | null
  ip_addresses: string | null
  url_domains: string | null
  personal_email_breadth: string | number | null
  personal_email_excluded_domains: string | null
  personal_email_max_recipients: string | number | null
  modified_date: string | null
  modified_by_id: string | number | null
  object_uuid: string | null
}

export interface DlpPolicySearchResponse {
  items: DlpPolicyExclusion[]
  row_count: number
  max_rows: number
  truncated: boolean
  query_duration_ms: number
  source_refreshed_at: string
  integration_status: string
}

export async function fetchDlpPolicyExclusions(): Promise<DlpPolicySearchResponse> {
  const response = await apiClient.get<DlpPolicySearchResponse>('/dlp-policy-search')
  return response.data
}

File: backend/app/schemas/dlp_policy_search.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class DlpPolicyExclusion(BaseModel):
    object_id: Any = None
    object_name: str | None = None
    object_description: str | None = None
    object_status: str | None = None
    rule_type: Any = None
    used_as: str
    policy_id: Any = None
    policy_name: str | None = None
    policy_active_status: Any = None
    policy_record_status: str | None = None
    user_patterns: str | None = None
    ip_addresses: str | None = None
    url_domains: str | None = None
    personal_email_breadth: Any = None
    personal_email_excluded_domains: str | None = None
    personal_email_max_recipients: Any = None
    modified_date: str | None = None
    modified_by_id: Any = None
    object_uuid: str | None = None


class DlpPolicySearchResponse(BaseModel):
    items: list[DlpPolicyExclusion]
    row_count: int
    max_rows: int
    truncated: bool
    query_duration_ms: int
    source_refreshed_at: datetime
    integration_status: str
